subiquity/models/tests/test_subiquity.py

```diff
@@ -227,3 +234,37 @@ def test_cloud_init_user_list_merge(self, run_cmd):
             cloud_init_config = model._cloud_init_config()
             self.assertEqual(len(cloud_init_config['users']), 1)
             self.assertEqual(cloud_init_config['users'][0]['name'], 'user2')
+
+    def test_cloud_init_files_emits_datasource_config_and_clean_script(self):
+        main_user = IdentityData(
+            username='mainuser',
+            crypted_password='sample_pass',
+            hostname='somehost')
+
+        model = self.make_model()
+        model.identity.add_user(main_user)
+        model.userdata = {}
+        expected_files = {
+            'etc/cloud/cloud.cfg.d/subiquity-disable-cloudinit-networking.cfg':
+                'network: {config: disabled}\n',
+            'etc/cloud/cloud.cfg.d/99-installer.cfg': re.compile('datasource:\n  None:\n    metadata:\n      instance-id: .*\n    userdata_raw: "#cloud-config\\\\ngrowpart:\\\\n  mode: \\\'off\\\'\\\\npreserve_hostname: true\\\\n\\\\\n'), # noqa
```

Comment:
Do you mind if we stuff this in a multiline string somewhere?
For the purpose of this test I'm less concerned about the contents of 99-installer.cfg and wouldn't mind looking elsewhere to find that.
Making it an "r" string would probably help the readability as well, if that works.

Reply:
Done and fixed up unit tests.